app/code/Magento/Sales/view/adminhtml/templates/order/create/form/address.phtml

```diff
@@ -89,11 +89,7 @@ endif; ?>
         <?= $block->getForm()->toHtml() ?>
 
         <div class="admin__field admin__field-option order-save-in-address-book">
-            <input name="<?= $block->getForm()->getHtmlNamePrefix() ?>[save_in_address_book]" type="checkbox"
-                   id="<?= $block->getForm()->getHtmlIdPrefix() ?>save_in_address_book"
-                   value="1"
-                   <?php if (!$block->getDontSaveInAddressBook() && $block->getAddress()->getSaveInAddressBook()): ?> checked="checked"<?php endif; ?>
```

Comment:
I think we should keep at least first part of the condition.
```
!$block->getDontSaveInAddressBook()
```
So we would make sure the saving billing address is turned off, when it is the same as shipping.